polar-core/src/sugar.rs

```diff
@@ -202,63 +275,248 @@ impl Namespaces {
         }
     }
 
-    fn get_related_type(&self, resource: &Term, relation: &Term) -> PolarResult<&Term> {
-        self.get_declaration(resource, relation)?.as_relation()
+    fn get_related_type(&self, relation: &Term, resource: &Term) -> PolarResult<&Term> {
+        self.get_declaration(relation, resource)?.as_relation()
     }
 
-    fn local_predicate_name(&self, resource: &Term, name: &Term) -> PolarResult<Symbol> {
-        Ok(self.get_declaration(resource, name)?.as_predicate())
+    fn local_declaration_to_rule_name(&self, name: &Term, resource: &Term) -> PolarResult<Symbol> {
+        Ok(self.get_declaration(name, resource)?.as_predicate())
     }
 
     fn cross_resource_predicate_name(
         &self,
-        resource: &Term,
-        relation: &Term,
         name: &Term,
+        relation: &Term,
+        resource: &Term,
     ) -> PolarResult<Symbol> {
-        let related_type = self.get_related_type(resource, relation)?;
-        self.local_predicate_name(related_type, name)
+        let related_type = self.get_related_type(relation, resource)?;
+        self.local_declaration_to_rule_name(name, related_type)
     }
 }
 
+impl KnowledgeBase {
```

Comment:
This method should either be implementing in `kb.rs` or this method should be changed to take a reference to the kb instead.